common/utils/src/test/scala/org/apache/spark/util/StructuredLoggingSuite.scala

```diff
@@ -45,39 +45,67 @@ abstract class LoggingSuiteBase extends AnyFunSuite // scalastyle:ignore funsuit
     val newContent = Files.readString(logFile.toPath)
     newContent.substring(content.length)
   }
-}
 
-class StructuredLoggingSuite extends LoggingSuiteBase {
-  private val className = this.getClass.getName.stripSuffix("$")
-  override def logFilePath: String = "target/structured.log"
+  def basicMsg: String = "This is a log message"
+
+  def msgWithMDC: LogEntry = log"Lost executor ${MDC(EXECUTOR_ID, "1")}."
+
+  def msgWithMDCAndException: LogEntry = log"Error in executor ${MDC(EXECUTOR_ID, "1")}."
```

Comment:
so I'm a little confused by this, it has basically just text in it like msgWithMDC so where is the "AndException" part?

Reply:
When logging with an exception, the output will contain the field "exception".

{"ts":"2024-04-01T09:43:18.892-0700","level":"ERROR","msg":"Error in executor 1.","context":{"executor_id":"1"},"exception":{"class":"java.lang.RuntimeException","msg":"OOM","stacktrace":"..."},"logger":"org.apache.spark.util.StructuredLoggingSuite"}

Reply:
Ok but the function itself doesn't have an exception in it, this just seems to be another string with MDC which is just different text then msgWithMDC  looking at where you use it I see the message copied into expectedPatternForMsgWithMDCAndException and then this passed into "Logging with MDC and Exception"...  why isn't this just called msgWithMDC2 since this function itself doesn't take an exception or add any functionality related to an exception

Reply:
Yes, we can make it `msgWithMDC2`, or reuse `msgWithMDC` and remove  `msgWithMDCAndException`.